src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,201 @@
+package com.internet.shop.dao.jdbc;
+
+import com.internet.shop.dao.UserDao;
+import com.internet.shop.exceptions.DataProcessingException;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Role;
+import com.internet.shop.model.User;
+import com.internet.shop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+@Dao
+public class UserDaoJdbcImpl implements UserDao {
+    @Override
+    public Optional<User> findByLogin(String login) {
+        String query = "SELECT * FROM users WHERE login = ? AND deleted = FALSE;";
+        User user = null;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, login);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                user = getUserFromResultSet(resultSet);
+            }
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't get user with login " + login, ex);
+        }
+
+        if (user != null) {
+            user.setRoles(getUserRolesFromResultSet(user.getId()));
+        }
+        return Optional.ofNullable(user);
+    }
+
+    @Override
+    public User create(User user) {
+        String query = "INSERT INTO users (user_name, login, password) VALUES (?, ?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query,
+                    Statement.RETURN_GENERATED_KEYS);
+            statement.setString(1, user.getName());
+            statement.setString(2, user.getLogin());
+            statement.setString(3, user.getPassword());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                user.setId(resultSet.getLong(1));
+            }
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't create new user", ex);
+        }
+
+        insertUserRoles(user);
+        return user;
```

Comment:
does your user has ids for roles, if yes, when have set them ?

Reply:
Added needed method